docs/proposals/backup-restore.md

```diff
@@ -1,131 +1,258 @@
+* `--insecure`: Allow connections to registries without valid TLS certificates.
+* `--registry-config <path>`: Path to the authentication configuration file for the registry.
+* `--username <string>`: Username for authenticating to the registry.
+* `--password <string>`: Password for authenticating to the registry.
+* `--password-stdin`: Read password from stdin.
+* `--identity-token <string>`: Use bearer token for authentication.
+* `--identity-token-stdin`: Read identity token from stdin.
+* `--ca-file <path>`: Path to custom CA certificate file.
+* `--cert-file <path>`: Path to client TLS certificate file.
+* `--key-file <path>`: Path to client private key file.
+* `--resolve <host:port:address[:address_port]>`: Customized DNS for registry.
+* `--debug`: Output debug logs (implies `--no-tty`).
+* `--distribution-spec string`: [Preview] set OCI distribution spec version and API option for target. Options: v1.1-referrers-tag, v1.1-referrers-api
+* `--no-tty`: Disable progress bars.
+
+### User Experience in the CLI
+
+The desired end-to-end user experience of using `oras backup` and `oras restore` to address the identified problems and support the outlined user scenarios is illustrated below.
+
+#### Offline Snapshot for Air-Gapped Environments
+
+Create a snapshot of a sample image `registry-a.k8s.io/kube-apiserver:v1` and its referrer (e.g. signature) for an air-gapped environment:
+
+```console
+oras backup registry-a.k8s.io/kube-apiserver:v1 --include-referrers --output airgap-snapshot.tar
 ```
 
-The backup command will also have the ability to write output to a new compressed tar file where the contents are in a single oci-layout
-format. For example:
+Upon success, the output will be:
 
-```bash
-oras backup --output ./mirror.tgz  registry.k8s.io/kube-apiserver-arm64:v1.31.0 registry.k8s.io/kube-controller-manager-arm64:v1.31.0
+```console
+Pulled 1 descriptor(s) from registry-a.k8s.io/kube-apiserver:v1
+Found 1 linked referrer(s)
+Included referrers in backup: application/vnd.cncf.notary.signature
+Exported backup to airgap-snapshot.tar
+Backup completed: 2 artifact(s) written
 ```
 
-If the output specified is an existing directory, the output will be written in that directory in OCI layout format.
-If the output specified is an existing file, it will be overwritten.
-If the output specified is neither a file or a directory, file output is assumed.
-The file name does NOT need to end in `tgz`, but file output will be compressed tar file.
+Transfer the `.tar` file to the air-gapped system via a secured channel. Restore the tarball from local to another registry:
 
-### oras restore
+```console
+oras restore registry-b.k8s.io/kube-apiserver:v1 --input airgap-snapshot.tar
+```
+
+Upon success, the output will be:
+
+```console
+Loaded 2 artifact(s) from airgap-snapshot.tar
+Pushed image to registry-b.k8s.io/kube-apiserver:v1
+Pushed linked referrer: sha256:78833f9c870...
```

Comment:
Do we want to output the artifact type of the referrer as well? Something like:
> Pushed referrer: application/vnd.cncf.notary.signature: sha256:78833f9c870...

Same for the backup command

Reply:
Good point. Added the artifact type to each referrer digest